### PR TITLE
Update Social Media Auth page with Twitter Signin

### PR DESCRIPTION
**Pull request by ndejesus1227**
I've tested this with both iOS and Android and it worked. Let me know if there's anything that should be changed

🔥

### CODE REVIEW

File: docs/auth/social-auth.md [hunk 1]
```diff
@@ -77,6 +77,38 @@ export const googleLogin = async () => {
...
 ## Twitter
+You can use react-native-twitter-signin for Twitter authentication. This library handles the flow of logging in a user and accesssing their `authToken` and `authTokenSecrent` by wrapping around the official Twitter library.
+
+The`react-native-twitter-signin` library allows us to login (using `RNTwitterSignIn`) which returns `authToken` and `authTokenSecret`
+
+Make sure you thorougly follow the instructions on setting up the necessary requirements to get react-native-twitter-signin working.
+
+
+```js
+import {NativeModules} from 'react-native}
```
Comment (Salakar) [inline docs/auth/social-auth.md]:
```suggestion
import {NativeModules} from 'react-native}';
```

File: docs/auth/social-auth.md [hunk 2]
```diff
@@ -77,6 +77,38 @@ export const googleLogin = async () => {
...
+import firebase from 'react-native-firebase';
+
+const {RNTwitterSignIn} = NativeModules;
+
+const TwittrKeys = {
```
Comment (Salakar) [inline docs/auth/social-auth.md]:
```suggestion
const TwitterKeys = {
```

File: docs/auth/social-auth.md [hunk 3]
```diff
@@ -77,6 +77,38 @@ export const googleLogin = async () => {
...
+   TWITTER_CONSUMER_KEY: 'PLACE_YOUR_TWITTER_CONSUMER_KEY_HERE'
+   TWITTER_CONSUMER_SECRET: 'PLACE_YOUR_TWITTER_CONSUMER_SECRET_HERE'
+}
+
+export const signinTwitter = async = () => { 
+    RNTwitterSignin.init(TwitterKeys.TWITTER_CONSUMER_KEY, TwitterKeys.TWITTER_CONSUMER_SECRET);
```
Comment (Salakar) [inline docs/auth/social-auth.md]:
```suggestion
    RNTwitterSignIn.init(TwitterKeys.TWITTER_CONSUMER_KEY, TwitterKeys.TWITTER_CONSUMER_SECRET);
```

File: docs/auth/social-auth.md [hunk 4]
```diff
@@ -77,6 +77,38 @@ export const googleLogin = async () => {
...
+
+
+```js
+import {NativeModules} from 'react-native}
+import firebase from 'react-native-firebase';
+
+const {RNTwitterSignIn} = NativeModules;
+
+const TwittrKeys = {
+   TWITTER_CONSUMER_KEY: 'PLACE_YOUR_TWITTER_CONSUMER_KEY_HERE'
```
Comment (Salakar) [inline docs/auth/social-auth.md]:
```suggestion
   TWITTER_CONSUMER_KEY: 'PLACE_YOUR_TWITTER_CONSUMER_KEY_HERE',
```

File: docs/auth/social-auth.md [hunk 5]
```diff
@@ -77,6 +77,38 @@ export const googleLogin = async () => {
...
+   TWITTER_CONSUMER_SECRET: 'PLACE_YOUR_TWITTER_CONSUMER_SECRET_HERE'
+}
```
Comment (Salakar) [inline docs/auth/social-auth.md]:
```suggestion
};
```

File: docs/auth/social-auth.md [hunk 6]
```diff
@@ -77,6 +77,38 @@ export const googleLogin = async () => {
...
+
+export const signinTwitter = async = () => { 
+    RNTwitterSignin.init(TwitterKeys.TWITTER_CONSUMER_KEY, TwitterKeys.TWITTER_CONSUMER_SECRET);
+    
+    try {
+    const result = await RNTwitterSignIn.logIn();
```
Comment (Salakar) [inline docs/auth/social-auth.md]:
```suggestion
      const result = await RNTwitterSignIn.logIn();
```

File: docs/auth/social-auth.md [hunk 7]
```diff
@@ -77,6 +77,38 @@ export const googleLogin = async () => {
...
+    
+    const credential = firebase.auth.TwitterAuthProvider.credential(result.authToken, result.authTokenSecret);
```
Comment (Salakar) [inline docs/auth/social-auth.md]:
```suggestion
      const credential = firebase.auth.TwitterAuthProvider.credential(result.authToken, result.authTokenSecret);
```

File: docs/auth/social-auth.md [hunk 8]
```diff
@@ -77,6 +77,38 @@ export const googleLogin = async () => {
...
+    
+    await firebase.auth().signInWithCredential(credential);
```
Comment (Salakar) [inline docs/auth/social-auth.md]:
```suggestion
      await firebase.auth().signInWithCredential(credential);
```

File: docs/auth/social-auth.md [hunk 9]
```diff
@@ -77,6 +77,38 @@ export const googleLogin = async () => {
 ```
 
 ## Twitter
+You can use react-native-twitter-signin for Twitter authentication. This library handles the flow of logging in a user and accesssing their `authToken` and `authTokenSecrent` by wrapping around the official Twitter library.
```
Comment (Salakar) [inline docs/auth/social-auth.md]:
```suggestion
You can use `react-native-twitter-signin` for Twitter authentication. This library handles the flow of logging in a user and accessing their `authToken` and `authTokenSecret` by wrapping around the official Twitter library.
```

File: docs/auth/social-auth.md [hunk 10]
```diff
@@ -77,6 +77,38 @@ export const googleLogin = async () => {
...
+
+The`react-native-twitter-signin` library allows us to login (using `RNTwitterSignIn`) which returns `authToken` and `authTokenSecret`
+
+Make sure you thorougly follow the instructions on setting up the necessary requirements to get react-native-twitter-signin working.
```
Comment (Salakar) [inline docs/auth/social-auth.md]:
```suggestion
Make sure you thoroughly follow the instructions on setting up the necessary requirements to get `react-native-twitter-signin` working.
```

File: docs/auth/social-auth.md (hunk 1 above)
Comment (ndejesus1227) [reply docs/auth/social-auth.md]: There's a random bracket at the end, it should be

import { NativeModules } from 'react-native';

Comment (Salakar) [reply docs/auth/social-auth.md]: Updated

**Pull request merged**